python/py-onnx/Portfile

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+
+github.setup        onnx onnx 1.6.0 v
+revision            0
+name                py-${github.project}
+categories-append   science
+fetch.type          git
+
+platforms           darwin
+supported_archs     noarch
+
+license             BSD
+
+maintainers         nomaintainer
+
+description         Open Neural Network Exchange
+
+long_description    Open Neural Network Exchange (ONNX) is an open\
+                    ecosystem that empowers AI developers to choose\
+                    the right tools as their project evolves. ONNX\
+                    provides an open source format for AI models, both\
+                    deep learning and traditional ML. It defines an\
+                    extensible computation graph model, as well as\
+                    definitions of built-in operators and standard\
+                    data types. Currently we focus on the capabilities\
+                    needed for inferencing (scoring).
+
+homepage            https://onnx.ai/
+
+# Support python versions.
+python.versions     27 35 36 37
+
+compiler.cxx_standard 2011
+
+if {${name} ne ${subport}} {
+
+    depends_build-append \
+        port:git \
+        port:cctools \
+        path:bin/cmake:cmake \
+        port:py${python.version}-protobuf3 \
+        port:py${python.version}-pytest-runner
+
+    depends_lib-append \
+        port:protobuf-c \
+        port:py${python.version}-future \
+        port:py${python.version}-numpy \
+        port:py${python.version}-setuptools \
+        port:py${python.version}-six
+
+    if { ${python.version} < 35 } {
+        depends_lib-append \
+            port:py${python.version}-typing \
+    }
+
+    if { ${python.version} >= 37 } {
```

Comment:
why is this only for Python 3.7 and above - upstream seems to include this dependence unconditionally?

Reply:
AFAIK, this port is only for 37+. I believe that upstream is incorrect to grab it unconditionally. Lower versions don’t exist in MacPorts anyway.